eis_toolkit/raster_processing/create_constant_raster.py

```diff
@@ -0,0 +1,186 @@
+        template_raster: An optional raster to use as a template for the output.
+        coord_west: The western coordinate of the output raster in [m].
+        coord_east: The eastern coordinate of the output raster in [m].
+        coord_south: The southern coordinate of the output raster in [m].
+        coord_north: The northern coordinate of the output raster in [m].
+        target_epsg: The EPSG code for the output raster.
+        target_pixel_size: The pixel size of the output raster.
+        raster_width: The width of the output raster.
+        raster_height: The height of the output raster.
+        nodata_value: The nodata value of the output raster.
+
+    Returns:
+        A tuple containing the output raster as a NumPy array and updated metadata.
+
+    Raises:
+        InvalidParameterValueException: Provide invalid input parameter.
+    """
+
+    if template_raster is not None:
+        out_array, out_meta = _create_constant_raster_from_template(constant_value, template_raster, nodata_value)
+    else:
+        if raster_height <= 0 or raster_width <= 0:
+            raise InvalidParameterValueException("Invalid raster extent provided.")
+        else:
+            if all(coords is not None for coords in [coord_west, coord_east, coord_south, coord_north]):
+                if not check_minmax_position((coord_west, coord_east)):
+                    raise InvalidParameterValueException("Invalid longitude values provided.")
+                elif not check_minmax_position((coord_south, coord_north)):
+                    raise InvalidParameterValueException("Invalid longitude values provided.")
+                else:
+                    out_array, out_meta = _create_constant_raster_from_bounds(
+                        constant_value,
+                        coord_west,
+                        coord_north,
+                        coord_east,
+                        coord_south,
+                        target_epsg,
+                        raster_width,
+                        raster_height,
+                        nodata_value,
+                    )
+            elif all(coords is not None for coords in [coord_west, coord_north]) and all(
+                coords is None for coords in [coord_east, coord_south]
+            ):
+                if target_pixel_size <= 0:
+                    raise InvalidParameterValueException("Invalid pixel size.")
+                else:
+                    out_array, out_meta = _create_constant_raster_from_origin(
+                        constant_value,
+                        coord_west,
+                        coord_north,
+                        target_epsg,
+                        target_pixel_size,
+                        raster_width,
+                        raster_height,
+                        nodata_value,
+                    )
+
+        constant_value = cast_scalar_to_int(constant_value)
+        nodata_value = cast_scalar_to_int(out_meta["nodata"])
```

Comment:
This comment relates to just code style, but I would reduce nesting here. The elifs and elses after raised exceptions could be removed. Here is one suggestion, but this is not a necessary modification:
```suggestion
    if template_raster is not None:
        out_array, out_meta = _create_constant_raster_from_template(constant_value, template_raster, nodata_value)

    elif all(coords is not None for coords in [coord_west, coord_east, coord_south, coord_north]):
        if raster_height <= 0 or raster_width <= 0:
            raise InvalidParameterValueException("Invalid raster extent provided.")
        if not check_minmax_position((coord_west, coord_east) or not check_minmax_position((coord_south, coord_north))):
            raise InvalidParameterValueException("Invalid coordinate values provided.")
        
        out_array, out_meta = _create_constant_raster_from_bounds(
            constant_value,
            coord_west,
            coord_north,
            coord_east,
            coord_south,
            target_epsg,
            raster_width,
            raster_height,
            nodata_value,
        )

    elif all(coords is not None for coords in [coord_west, coord_north]) and all(
        coords is None for coords in [coord_east, coord_south]
    ):
        if raster_height <= 0 or raster_width <= 0:
            raise InvalidParameterValueException("Invalid raster extent provided.")
        if target_pixel_size <= 0:
            raise InvalidParameterValueException("Invalid pixel size.")

        out_array, out_meta = _create_constant_raster_from_origin(
            constant_value,
            coord_west,
            coord_north,
            target_epsg,
            target_pixel_size,
            raster_width,
            raster_height,
            nodata_value,
        )

    else:
        raise InvalidParameterValueException("Suitable parameter values were not provided for any of the 3 methods.")

    constant_value = cast_scalar_to_int(constant_value)
    nodata_value = cast_scalar_to_int(out_meta["nodata"])
```